core/arch/arm/plat-mediatek/main.c

```diff
@@ -0,0 +1,571 @@
+#include <pm_debug.h>
+
+#include <stdint.h>
+#include <string.h>
+
+#include <drivers/gic.h>
+#include <drivers/s8250_uart.h>
+#include <sm/sm.h>
+#include <sm/tee_mon.h>
+
+#include <util.h>
+
+#include <arm.h>
+#include <kernel/thread.h>
+#include <kernel/panic.h>
+#include <trace.h>
+#include <kernel/misc.h>
+#include <kernel/tee_time.h>
+#include <mm/tee_pager.h>
+#include <mm/core_mmu.h>
+#include <mm/tee_mmu_defs.h>
+#include <mm/tee_mmu.h>
+#include <mm/tee_mm.h>
+#include <utee_defines.h>
+#include <tee/tee_cryp_provider.h>
+#include <tee/entry.h>
+#include <tee/arch_svc.h>
+#include <console.h>
+#include <malloc.h>
+#include "plat_tee_func.h"
+
+#include <assert.h>
+
+#define PADDR_INVALID		0xffffffff
+
+#ifndef CFG_WITH_LPAE
+/* Main MMU L1 table for teecore */
+static uint32_t main_mmu_l1_ttb[TEE_MMU_L1_NUM_ENTRIES]
+	__attribute__((section(".nozi.mmu.l1"),
+		       aligned(TEE_MMU_L1_ALIGNMENT)));
+static uint32_t main_mmu_l2_ttb[TEE_MMU_L2_NUM_ENTRIES]
+	__attribute__((section(".nozi.mmu.l2"),
+		       aligned(TEE_MMU_L2_ALIGNMENT)));
+
+/* MMU L1 table for TAs, one for each Core */
+static uint32_t main_mmu_ul1_ttb[CFG_NUM_THREADS][TEE_MMU_UL1_NUM_ENTRIES]
+	__attribute__((section(".nozi.mmu.ul1"),
+		      aligned(TEE_MMU_UL1_ALIGNMENT)));
+#endif
+
+extern uint8_t __text_init_start[];
+extern uint8_t __data_start[];
+extern uint8_t __data_end[];
+extern uint8_t __bss_start[];
+extern uint8_t __bss_end[];
+extern uint8_t __init_start[];
+extern uint8_t __init_size[];
+extern uint8_t __heap1_start[];
+extern uint8_t __heap1_end[];
+extern uint8_t __heap2_start[];
+extern uint8_t __heap2_end[];
+extern uint8_t __pageable_part_start[];
+extern uint8_t __pageable_part_end[];
+extern uint8_t __pageable_start[];
+extern uint8_t __pageable_end[];
+
+static void main_fiq(void);
+#if defined(CFG_WITH_ARM_TRUSTED_FW)
+/* Implemented in assembly, referenced in this file only */
+uint32_t cpu_on_handler(uint32_t a0, uint32_t a1);
+
+static uint32_t main_cpu_off_handler(uint32_t a0, uint32_t a1);
+static uint32_t main_cpu_suspend_handler(uint32_t a0, uint32_t a1);
+static uint32_t main_cpu_resume_handler(uint32_t a0, uint32_t a1);
+static uint32_t main_system_off_handler(uint32_t a0, uint32_t a1);
+static uint32_t main_system_reset_handler(uint32_t a0, uint32_t a1);
+#elif defined(CFG_WITH_SEC_MON)
+static uint32_t main_default_pm_handler(uint32_t a0, uint32_t a1);
+#else
+#error Platform must use either ARM_TRUSTED_FW or SEC_MON
+#endif
+
+static const struct thread_handlers handlers = {
+	.std_smc = plat_tee_entry,
+	.fast_smc = plat_tee_entry,
+	.fiq = main_fiq,
+	.svc = tee_svc_handler,
+	.abort = tee_pager_abort_handler,
+#if defined(CFG_WITH_ARM_TRUSTED_FW)
+	.cpu_on = cpu_on_handler,
+	.cpu_off = main_cpu_off_handler,
+	.cpu_suspend = main_cpu_suspend_handler,
+	.cpu_resume = main_cpu_resume_handler,
+	.system_off = main_system_off_handler,
+	.system_reset = main_system_reset_handler,
+#elif defined(CFG_WITH_SEC_MON)
```

Comment:
CFG_WITH_SEC_MON is no more needed. Now, this either CFG_WITH_ARM_TRUSTED_FW is defined, either it is not - later case means the old CFG_WITH_SEC_MON was defined.